build.gradle: `classifier` property of `Jar` is deprecated and removed in Gradle 8.0

### Description

Gradle 8.0 has removed the deprecated property `classifier` of `Jar`. This is last seen in [Gradle 7.6 docs](https://docs.gradle.org/7.6/dsl/org.gradle.api.tasks.bundling.Jar.html#org.gradle.api.tasks.bundling.Jar:classifier).

### Steps to Reproduce (Bug)

(Extracted from https://github.com/Homebrew/homebrew-core/pull/123460; logs could be found at https://github.com/Homebrew/homebrew-core/actions/runs/4199791128/jobs/7285175571#step:6:338.)

```
==> gradle installDist
  Picked up _JAVA_OPTIONS: -Duser.home=/Users/brew/Library/Caches/Homebrew/java_cache
  
  Welcome to Gradle 8.0!
  
  For more details see https://docs.gradle.org/8.0/release-notes.html
  
  Starting a Gradle Daemon (subsequent builds will be faster)
  Configuration on demand is an incubating feature.
  > Task :buildSrc:compileJava
  > Task :buildSrc:compileGroovy NO-SOURCE
  > Task :buildSrc:processResources NO-SOURCE
  > Task :buildSrc:classes
  > Task :buildSrc:jar
  
  FAILURE: Build failed with an exception.
  
  * Where:
  Build file '/private/tmp/teku-20230216-44963-1h2407x/build.gradle' line: 127
  
  * What went wrong:
  A problem occurred evaluating root project 'teku'.
  > Could not set unknown property 'classifier' for task ':sourcesJar' of type org.gradle.api.tasks.bundling.Jar.
  
  * Try:
  > Run with --stacktrace option to get the stack trace.
  > Run with --info or --debug option to get more log output.
  > Run with --scan to get full insights.
  
  * Get more help at https://help.gradle.org/
  
  BUILD FAILED in 26s
  2 actionable tasks: 2 executed
```

**Expected behavior:** teku to be successfully built

**Actual behavior:** See commant output above

**Frequency:** N/A

### Versions (Add all that apply)

* Software version: 23.2.0

This is specific to Gradle 8.0 and should be reproducible on all platforms; the following do not apply.

* Java version: [`java -version`]
* OS Name & Version: [`cat /etc/*release`]
* Docker Version: [`docker version`]
* Cloud VM, type, size: [Amazon Web Services I3-large]

--- a/build.gradle
+++ b/build.gradle
@@ -976,3 +976,4 @@ task cloudsmithUpload {
     }
   }
 }
+

--- a/build.gradle
+++ b/build.gradle
@@ -124,7 +124,7 @@ allprojects {
   version = rootProject.version
 
   task sourcesJar(type: Jar, dependsOn: classes) {
-    classifier = 'sources'
+    archiveClassifier = 'sources'
     from sourceSets.main.allSource
     jar.reproducibleFileOrder = true
     jar.preserveFileTimestamps = false
@@ -976,4 +976,3 @@ task cloudsmithUpload {
     }
   }
 }
-

